Define PSA cryptography repository (#6612)
Mbed-TLS/mbedtls: docs/architecture/psa-crypto-repository.md

PSA Cryptography repository
===========================
## Introduction

The PSA Cryptography repository contains a reference implementation of the [PSA Cryptography API specification](https://armmbed.github.io/mbed-crypto/psa/#application-programming-interface).

## Requirements

> * The PSA Cryptography repository exposes as public interface the cryptographic interface defined in the PSA Cryptography API specification and solely this interface.

Reviewer: It is possible we might end up exposing some extra bits (PKCS maybe), but these will be small exceptional cases. We definitely won't expose the legacy interface.

Reviewer 2: Which PKCS are you talking about?

“We definitely won't expose the legacy interface” vs “we might end up exposing some extra bits” — well, which one is it? That's a major design decision.

Author: I've changed that, including the unified driver interface and the on-going PSA crypto extensions.

> * The PSA Cryptography repository provides a way to build and test a C static and/or shared library exposing completely or partially the PSA Cryptography API.

Reviewer: ```suggestion
* The PSA Cryptography repository provides a way to independently build and test a C static and/or shared library exposing completely or partially the PSA Cryptography API, without relying on the Mbed TLS repository.
```

> * The PSA Cryptography repository provides a way to define the parts of the PSA Cryptography API exposed by the built C library.

Reviewer: What does "provides a way to define..." mean?

Author: I am changing to "provides a configuration mechanism to define ...".

* The PSA Cryptography repository is derived from the Mbed TLS repository. No cryptographic development activities as such will occur on the PSA Cryptography repository.
* The PSA Cryptography repository is derived from the Mbed TLS repository but it does not mean that all its content comes from Mbed TLS. It may contain a marginal number of files on its own.

> * The PSA Cryptography repository should be able to evolve to be the development repository of the PSA Cryptography reference implementation.

Reviewer: ```suggestion
* The PSA Cryptography repository must be able to evolve to be the development repository of the PSA Cryptography reference implementation.
```

> * The update of the PSA Cryptography repository from the Mbed TLS repository should be automated. It is expected that the automation itself evolves with the evolutions of the Mbed TLS repository but the less the best. The trigger of the updates may not be automated.

Reviewer: ```suggestion
* The update of the PSA Cryptography repository from the Mbed TLS repository should be automated and done at a reasonably short cadence (i.e., at least monthly). It is expected that the automation itself evolves with the evolutions of the Mbed TLS repository but the less the better. The trigger of the updates may or may not be automated.
```
